### PR TITLE
pax: init at 0.4.0

### PR DESCRIPTION
**Pull request by klntsky**
###### Motivation for this change

Pax is a JS bundler, see https://pax.js.org/

###### Things done
- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Not sure if including `Cargo.lock` through Github Gists is a good idea, but it is done the same way at [cargo-fuzz](https://github.com/NixOS/nixpkgs/blob/92a047a6c4d46a222e9c323ea85882d0a7a13af8/pkgs/development/tools/rust/cargo-fuzz/default.nix).

### CODE REVIEW

**Comment by 8084** [comment]
In AUR it is named [pax-rs](https://aur.archlinux.org/packages/pax-rs/), is this name OK?

**Comment by 8084** [comment]
By the way, the binary is named `px` for short, not `pax`. I hope it will not cause conflicts.

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -2312,7 +2312,11 @@
     email = "tierpluspluslists@gmail.com";
     name = "Karn Kallio";
   };
-
+  klntsky = {
```
Comment (veprbl) [inline maintainers/maintainer-list.nix]: You can use your github name here.
```nix
{
  "8084" = {
    # ...
  };
}
```

```nix
{
  maintainers = [ maintainers."8084" ];
}
```

File: pkgs/development/tools/pax/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{ stdenv, fetchFromGitHub, fetchurl, rustPlatform, runCommand } :
+with rustPlatform;
+
+buildRustPackage rec {
+  name = "pax-${version}";
```
Comment (veprbl) [inline pkgs/development/tools/pax/default.nix]:
```suggestion
  name = "pax-rs-${version}";
```

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -8929,6 +8929,8 @@ in
 
   patchelfUnstable = lowPrio (callPackage ../development/tools/misc/patchelf/unstable.nix { });
 
+  pax = callPackage ../development/tools/pax { };
```
Comment (veprbl) [inline pkgs/top-level/all-packages.nix]:
```suggestion
  pax-rs = callPackage ../development/tools/pax-rs { };
```

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (8084) [reply maintainers/maintainer-list.nix]: Thanks, I'd rather prefer to leave it as it is.

**Review by veprbl — APPROVED** [review]

**Comment by veprbl** [comment]
@GrahamcOfBorg build pax-rs

**Comment by veprbl** [comment]
@8084 Congratulations on the first contribution!

**Pull request merged**